### DISCUSSION TITLE
Can a SED be told to find a different route?

### DISCUSSION THREAD

**Discussion by ksathara** [openthread/openthread#10312] [Q&A]
Hi,
Is there a way for the SED to renegotiate a different route under application fw? 
I have a parent router in the middle which is in some error state, but the SED keep on connecting to it. Suppose I have an app level handshake  between the SED and the final router, so I can detect there is some issue in the path. What I want to know is, is it possible for the SED to drop the current connection and renegotiate a new path?
Cheers,
Kaushalya

**Comment by jwhui** [comment]
A SED can detach and reattach. However, the Thread Specification defines how a SED choses which neighboring device to select as a parent.

That said, OpenThread does support [`macfilter`](https://openthread.io/reference/cli/commands#macfilter_addr_add) capability to allow/deny specific neighboring devices based on their MAC address. But this feature is only intended for testing.

**Reply by ksathara** [reply]
Hi,
I have tried following using the CLI

1. detach
2. thread stop
3. thread start
This causes the SED to connect back but it seem to connect to the same router. I have tried witing 240 sec betwenn steps 2 and 3 same result. Only thing is every time the SED gets a new rloc.
Any way to make the SED connect a different router?
Thanks,
Kaushalya

**Reply by jwhui** [reply]
As mentioned above: OpenThread does support [`macfilter`](https://openthread.io/reference/cli/commands#macfilter_addr_add) capability to allow/deny specific neighboring devices based on their MAC address. But this feature is only intended for testing.
